jetstream/consumer_config.go

```diff
@@ -20,85 +20,243 @@ import (
-		MaxRequestBatch    int             `json:"max_batch,omitempty"`
-		MaxRequestExpires  time.Duration   `json:"max_expires,omitempty"`
-		MaxRequestMaxBytes int             `json:"max_bytes,omitempty"`
-
-		// Inactivity threshold.
+		// Name is an optional name for the consumer. If not set, one is
+		// generated automatically.
+		Name string `json:"name,omitempty"`
+
+		// Durable is an optional durable name for the consumer. If both Durable
+		// and Name are set, they have to be equal. If Durable is set and Name
+		// is not, Name is set to Durable. Unless InactiveThreshold is set, a
+		// durable consumer will not be cleaned up automatically.
+		Durable string `json:"durable_name,omitempty"`
+
+		// Description provides an optional description of the consumer.
+		Description string `json:"description,omitempty"`
+
+		// DeliverPolicy defines from which point to start delivering messages
+		// from the stream. Defaults to DeliverAllPolicy.
+		DeliverPolicy DeliverPolicy `json:"deliver_policy"`
+
+		// OptStartSeq is an optional sequence number from which to start
+		// message delivery. Only applicable when DeliverPolicy is set to
+		// DeliverByStartSequencePolicy.
+		OptStartSeq uint64 `json:"opt_start_seq,omitempty"`
+
+		// OptStartTime is an optional time from which to start message
+		// delivery. Only applicable when DeliverPolicy is set to
+		// DeliverByStartTimePolicy.
+		OptStartTime *time.Time `json:"opt_start_time,omitempty"`
+
+		// AckPolicy defines the acknowledgement policy for the consumer.
+		// Defaults to AckExplicitPolicy.
+		AckPolicy AckPolicy `json:"ack_policy"`
+
+		// AckWait defines how long the server will wait for an acknowledgement
+		// before resending a message. Defaults to 30 seconds.
+		AckWait time.Duration `json:"ack_wait,omitempty"`
+
+		// MaxDeliver defines the maximum number of delivery attempts for a
+		// message. Applies to any message that is re-sent due to ack policy.
+		// Defaults to -1 (unlimited).
+		MaxDeliver int `json:"max_deliver,omitempty"`
+
+		// BackOff specifies the optional back-off intervals for retrying
+		// message delivery after a failed acknowledgement. It overrides
+		// AckWait.
+		//
+		// The number of intervals specified must be lower or equal to
+		// MaxDeliver. If the number of intervals is lower, the last interval is
+		// used for all remaining attempts.
+		BackOff []time.Duration `json:"backoff,omitempty"`
+
+		// FilterSubject can be used to filter messages delivered form the
+		// stream. It has to overlap with the subjects defined on the stream.
+		// FilterSubject is exclusive with FilterSubjects.
+		FilterSubject string `json:"filter_subject,omitempty"`
+
+		// ReplayPolicy defines the rate at which messages are sent to the
+		// consumer. If ReplayOriginalPolicy is set, messages are sent at the
+		// same rate as they were stored in the stream. If ReplayInstantPolicy
+		// is set, messages are sent as fast as possible. Defaults to
+		// ReplayInstantPolicy.
```

Comment:
Changed to:
```go
		// ReplayPolicy defines the rate at which messages are sent to the
		// consumer. If ReplayOriginalPolicy is set, messages are sent in the
		// same intervals in which they were stored on stream. This can be used
		// e.g. to simulate production traffic in development environments. If
		// ReplayInstantPolicy is set, messages are sent as fast as possible.
		// Defaults to ReplayInstantPolicy.
```